internal/driver/driver.go

```diff
@@ -1,36 +1,73 @@
 package driver
 
 import (
+	"errors"
 	"log"
 
-	"github.com/seemsod1/api-project/internal/config"
+	"github.com/kelseyhightower/envconfig"
+	"github.com/seemsod1/api-project/internal/models"
 
 	"gorm.io/driver/postgres"
 	"gorm.io/gorm"
 )
 
-// DB is a struct that embeds the gorm.DB
-type DB struct {
+// DatabaseConfig is a struct that holds the database configuration
+type DatabaseConfig struct {
+	DSN string
+}
+
+func NewDatabaseConfig() *DatabaseConfig {
+	var dbConfig DatabaseConfig
+	if err := envconfig.Process("DB", &dbConfig); err != nil {
+		log.Fatal(err)
+	}
+	return &dbConfig
+}
+
+func (dc *DatabaseConfig) Validate() bool {
+	return dc.DSN != ""
+}
+
+// GORMDriver is a struct that embeds the gorm.DB
+type GORMDriver struct {
 	SQL *gorm.DB
```

Comment:
Consider changing `SQL` to something to something more descriptive and straightforwards, such as `DB` or `GormDB`